department-of-veterans-affairs/caseflow
# Don't fetch document count if we don't need it

Only request document count from BE if
  a. we don't already have it in the state or
  b. the count we have comes from Documents when we want the vbms count or
     vice versa

Resolves #8981 

### Description
Before requesting document count, check to see if we already have it stored and it came from the correct source (caseflow Documents table vs VBMS)

### Acceptance Criteria
- [X] Fetch document count if the document count does not exist in the redux store
- [X]  Fetch document count when visiting a user's queue if value in redux store does not come from caseflow Documents table
- [X]  Fetch document count when visiting the case details page if value in redux store does not come from VBMS
- [X]  Do not fetch document count if value in redux store comes from VBMS

### Testing Plan
1. Go to a queue with reader links and multiple tabs
2. Switch to a different tab and back to the original one
3. Confirm no requests were made to document_count (reader link should not re-render)
